lambdas/search/index.py

```diff
@@ -50,23 +51,51 @@ def lambda_handler(request):
     """
 
     action = request.args.get('action')
-    indexes = request.args.get('index')
-    terminate_after = os.getenv('MAX_DOCUMENTS_PER_SHARD')
-
-    if action == 'search':
+    user_body = request.args.get('body', {})
+    user_fields = request.args.get('fields', [])
+    user_indexes = request.args.get('index', "")
+    user_size = request.args.get('size', DEFAULT_SIZE)
+    user_source = request.args.get('_source', [])
+    terminate_after = None  # see if we can skip os.getenv('MAX_DOCUMENTS_PER_SHARD')
+
+    if action == 'packages':
+        query = request.args.get('query', '')
+        body = user_body or {
+            "query": {
+                "query_string": {
+                    "analyze_wildcard": True,
+                    "lenient": True,
+                    "query": query,
+                    # see enterprise/**/bucket.py for mappings
+                    "fields": user_fields or [
+                        # package
+                        'comment', 'handle', 'handle_text^2', 'metadata_string', 'tags'
+                    ]
```

Comment:
@akarve we have to group the results by handle

Reply:
Sure, you can do the groupby (or aggregations) in the body right?

Reply:
oh, i can add a default for that sure but i'm letting you control the body for this action so you can sort/etc.

Reply:
oh i've just noticed that the whole body can be customized -- it should work